Converts PDAs to modular computers

comma opened the pull request:
Second season of #21033
:cl:mkalash
rscadd: PDAs are now modular computers.
/:cl:

closes #20373
closes #19956
closes #19640
closes #19011
closes #19002
closes #18588
closes #18194
closes #18000
closes #17353
closes #16689
closes #16515
closes #16342
closes #16115
closes #15379
closes #14747
closes #13990

mustafakalash committed:
Converts PDAs to modular computers.

mustafakalash committed:
Generalizes PDA code up to modular_computer.

comma committed:
Deuglifies email client a bit

Also fixes a runtime
And removes that editorbr crap, proper linebreaks are inserted now

comma committed:
Fixes client constantly logging out

mustafakalash commented:
I requested that my code not be merged. @xales

comma commented:
@xales fyi I knew and decided to carry on anyway.
It's a damn big and good change, already completed.
Seemed like a waste of code over whatever happened there in the chats.

Sunbeamstress commented:
This is it. This is the final boss of PRs. Time to use those stat-boost potions and elixirs you stockpiled.

Sbotkin commented:
So... no PDA messages anymore?

comma commented:
Yeah, emails are used as their replacement. Email program had tons of improvements to be not painful to use (if you have ID in PDA you can just login without typing email/pass, you get the messages with reply link to your chat like with PDAs, you can pick recipients from the list instead fo typing manually)

Minijar commented:
Thank god the login was solved, it was a pain in the arse to type that every time.

Sbotkin commented:
What handles the emails now? Will they receive dedicated server like the PDA one?

comma commented:
NTnet server I believe, bring down internets (or disable in admin program I dunno) and no emails

atlantiscze commented:
Right now it goes through NTNet relay, as pretty much everything that uses modular computer networking.

babydoll commented:
Pulled a branch, few things:

1. Opening a program, minimizing it, and then attempting to load another locks you out of the original one with the error about maximum CPU load - occurs on all modular computers, not happening on live (the number you can open depending on the CPU)
2. Remove ID hotkey didn't seem to make the transition (alt-click I believe)
3. Deleting autorun.dat doesn't remove the autorun buttons like I'd expect it to
4. Icon-wise looking pretty good, but the 'off' screen is just a grey field and could use a screensaver or something like the consoles do. I'll probably do a beauty pass if this is merged.

PsiOmegaDelta commented:
Closing as per original author request.

Minijar commented:
Are we never allowed to have modular PDAs now then because some sourpuss doesn't want his work used? Or must it be remade entirely to satisfy a single individual who's essentially left the community?

atlantiscze commented:
@sabiram 

2. I don't remember there being a remove ID hotkey for modular computers. Alt click is already reserved for toggle open/closed (used by laptops right now)

3. Not a bug. The autorun file only tells the system what program to load. Buttons merely modify the contents of this file so you don't have to do it manually. Not related to this PR as it was like this even before.

babydoll commented:
> I don't remember there being a remove ID hotkey for modular computers. Alt click is already reserved for toggle open/closed (used by laptops right now)

It was for PDAs, which these are replacing. These aren't laptops so the alt-click should still work, it just wasn't copied over.

babydoll committed:
Icon tweaks modcomputer PDA

babydoll commented:
@comma 
https://github.com/comma/Baystation12/pull/31

comma committed:
QoL tweaks

Reply button now opens the email client window
Can send messages without subject, will put 'No subject' there.
Fixed brs not being replaced properly when editing again
Added alt-click shortcut for popping ID out
No longer getting older message when responding to things.

babydoll commented:
Reposting from other branch:

> few notes on email:

>email program still needs some work. honestly just make it so it autofills info from mind, if possible. a lot of people dont use their ID in pda so its a pain in the neck having to put it in, log in, look at messages, then remove ID with the verb

>also email program: when you edit the message it converts new lines to [br] which is fine, but it makes it annoying to edit a message

>i'm not sure if the quoting thing is necessary. i prefer the old system in this regard

comma committed:
Merge fixes

comma committed:
Removes background autorun

Caused issues, works fine without it

comma commented:
Okay everythin sans 'fill from mind' is done, might look into that but otherwise it's done.

comma committed:
Makes login bit easier

It will try to grab ID off mob if none is inserted

comma committed:
Updates test thingies

Techhead0 commented on code/_helpers/global_access.dm:
```diff
@@ -2010,7 +2006,7 @@
 	"Master",
 	"OOClog",
-	"PDAs",
+	"SKILLS",
```
Should this be here?

Techhead0 commented on code/modules/clothing/under/accessories/badges.dm:
```diff
@@ -86,15 +86,15 @@
-			var/obj/item/device/pda/pda = O
-			id_card = pda.id
+			var/obj/item/modular_computer/computer = O
+			id_card = computer.GetIdCard()
```
```
var/obj/item/weapon/card/id/id_card = O.GetIdCard()
```
IDs return themselves.

Techhead0 commented on code/modules/mob/living/simple_animal/friendly/spiderbot.dm:
```diff
@@ -119,10 +119,10 @@
-			var/obj/item/device/pda/pda = O
-			id_card = pda.id
+			var/obj/item/modular_computer/computer = O
+			id_card = computer.GetIdCard()
```
Same `GetIdCard()` change.

Techhead0 reviewed (commented):
Read through it. Nothing major, few minor things.
